pkg/gds/models/v1/models.go

```diff
@@ -45,6 +46,27 @@ func SetAdminVerificationToken(vasp *pb.VASP, token string) (err error) {
 	return nil
 }
 
+// UpdateVerificationStatus changes the verification state of a VASP and appends a new
+// entry to the audit log on the extra.
+func UpdateVerificationStatus(vasp *pb.VASP, state pb.VerificationState, description string, source string) (err error) {
+	// Append a new entry to the audit log.
+	entry := &AuditLogEntry{
+		Timestamp:     time.Now().Format(time.RFC3339),
+		PreviousState: vasp.VerificationStatus,
+		CurrentState:  state,
+		Description:   description,
+		Source:        source,
+	}
+	if err := AppendAuditLog(vasp, entry); err != nil {
+		return err
+	}
+
+	// Set the new state on the VASP.
+	vasp.VerificationStatus = state
```

Comment:
I have a slight preference that the verification status gets changed before the audit log is appended, just in case we typo the previous state guard in `AppendAuditLog` but it's not a big deal.